### ISSUE TITLE
pwncat-cs installation issue (page 54)

### ISSUE THREAD

**Issue by Dougoll** [dolevf/Black-Hat-Bash#35]
Hi guys!

Currently reading your great book and setuping the lab. I am having issues installing pwncat-cs. Especially after a dist-upgrade. Tried to install with a python virtual environment, to no avail. Do you have any hints, a new install procedure or another similar tool to use?

Thank you very much!

**Comment by dolevf** [comment]
Can you share what error are you seeing?

**Comment by Dougoll** [comment]
![After_distro_install](https://github.com/user-attachments/assets/548f3199-d3dc-4058-8c58-dc47079bd3cd)
![after_distro_install_with_pipx](https://github.com/user-attachments/assets/40dac588-5855-4319-bae2-0057494d8b8b)

**Comment by dolevf** [comment]
If the tool itself is working and the exercises work ok, then it should be fine.

**Comment by Dougoll** [comment]
Alright! I'll let you know if I run into issues when I get to the pwncat-cs exercises.  Thanks for your time!

**Comment by awesomeandrey** [comment]
FYI

@Dougoll @dolevf I had to follow instructions from https://github.com/calebstewart/pwncat/issues/289 to overcome the installation issue.

**Comment by Dougoll** [comment]
Tested on latest version of Kali linux, this is how I have been able to make it work : 

1, Install build dependencies
sudo apt update
sudo apt install -y build-essential libssl-dev zlib1g-dev \
libncurses5-dev libncursesw5-dev libreadline-dev libsqlite3-dev \
libgdbm-dev libdb5.3-dev libbz2-dev libexpat1-dev liblzma-dev tk-dev \
wget curl git

2. Download and build python 3.11
cd /usr/src
sudo wget https://www.python.org/ftp/python/3.11.7/Python-3.11.7.tgz
sudo tar xvf Python-3.11.7.tgz
cd Python-3.11.7
sudo ./configure --enable-optimizations
sudo make -j$(nproc)
sudo make altinstall

3. Confirm Installation
python3.11 --version

4. Setup virtual environnement for pwncat-cs
python3.11 -m venv ~/pwncat-env
source ~/pwncat-env/bin/activate
pip install --upgrade pip
pip install pwncat-cs

5. Creating a sym link
sudo ln -s ~/pwncat-env/bin/pwncat-cs /usr/local/bin/pwncat-cs
